src/v/archival/scrubber.cc

```diff
@@ -70,33 +73,196 @@ ss::future<scrubber::purge_result> scrubber::purge_partition(
+        const auto manifest_delete_result = co_await _api.delete_object(
+          bucket,
+          cloud_storage_clients::object_key(*legacy_manifest_path),
+          legacy_manifest_delete_rtc);
+        if (manifest_delete_result != upload_result::success) {
+            ++retryable_failure;
+        }
+    }
+
+    // If any of the purges resulted in a retryable failure, map the entire
+    // operation to a retryable failure. Otherwise, if there were any permanent
+    // failures map to permanent failure.
+    if (retryable_failure > 0) {
+        vlog(
+          archival_log.info,
+          "Retryable failures encountered while purging partition {}. Will "
+          "retry ...",
+          ntp);
+
+        co_return purge_result{
+          .status = purge_status::retryable_failure, .ops = ops_performed};
+    } else if (permanent_failure > 0) {
+        vlog(
+          archival_log.error,
+          "Permanent failures encountered while purging partition {}. Giving "
+          "up ...",
+          ntp);
+
+        co_return purge_result{
+          .status = purge_status::permanent_failure, .ops = ops_performed};
+    } else {
+        vlog(
+          archival_log.info,
+          "Finished erasing partition {} from object storage in {} requests",
+          ntp,
+          ops_performed);
+
+        co_return purge_result{
+          .status = purge_status::success, .ops = ops_performed};
+    }
+}
+
+ss::future<std::optional<scrubber::collected_manifests>>
+scrubber::collect_manifest_paths(
+  const cloud_storage_clients::bucket_name& bucket,
+  model::ntp ntp,
+  model::initial_revision_id remote_revision,
+  retry_chain_node& parent_rtc) {
+    retry_chain_node collection_rtc(
+      ss::lowres_clock::now() + 5s, 1s, &parent_rtc);
 
     cloud_storage::partition_manifest manifest(ntp, remote_revision);
-    auto [manifest_get_result, manifest_fmt]
-      = co_await _api.try_download_partition_manifest(
-        bucket, manifest, manifest_rtc, true);
+    auto path = manifest.get_manifest_path(
+      cloud_storage::manifest_format::serde);
+
+    std::string_view base_path{path().native()};
+
+    vassert(
+      base_path.ends_with(serde_extension)
+        && base_path.length() > serde_extension.length(),
+      "Generated manifest path should end in .bin");
 
-    // save path here since manifest will get moved out
-    auto manifest_path = manifest.get_manifest_path(manifest_fmt);
+    base_path.remove_suffix(serde_extension.length());
+    auto list_result = co_await _api.list_objects(
+      bucket,
+      collection_rtc,
+      cloud_storage_clients::object_key{std::filesystem::path{base_path}});
+
+    if (list_result.has_error()) {
+        vlog(
+          archival_log.warn,
+          "Failed to collect manifests to scrub partition {}",
+          ntp);
+
+        co_return std::nullopt;
+    }
+
+    collected_manifests collected{};
+    collected.spillover.reserve(list_result.value().contents.size());
```

Comment:
Can you think about the edge case of someone writing 1GiB/s to a single partition for 3 years?  Would the size of this vector become large enough to exceed 128KiB?

(this is okay to fix in a followup, but needs consideration)